elyra/util/operation_parser.py

```diff
@@ -14,66 +14,60 @@
 # limitations under the License.
 #
 
-import asyncio
 import json
 import nbformat
-import os
 import re
 
 from abc import ABC, abstractmethod
 from jupyter_server.base.handlers import APIHandler
 from tornado import web
-from traitlets.config import SingletonConfigurable
-from typing import Any, Type, TypeVar, List
+from typing import Any, Type, TypeVar, List, Dict
 from ..util.http import HttpErrorMixin
 
 
 class OperationParserHandler(HttpErrorMixin, APIHandler):
     """Handler to expose REST API to parse envs from a File"""
 
     @web.authenticated
-    async def get(self):
+    async def post(self):
         msg_json = dict(title="Operation not supported.")
         self.write(msg_json)
         self.flush()
 
     @web.authenticated
-    async def post(self, *args, **kwargs):
-        payload = self.get_json_body()
+    async def get(self, path=''):
+        path = path or ''
 
-        self.log.debug("Parsing environmental variables from notebook: %s", payload['file_path'])
-        self.log.debug("JSON payload: %s", json.dumps(payload, sort_keys=True, indent=2, separators=(',', ': ')))
+        # TODO: Query and verify file type
+        # type = self.get_query_argument('type', default=None)
+        # if type not in {None, 'file', 'notebook'}:
+        #     raise web.HTTPError(400, u'Type %r is invalid' % type)
 
-        file_path = payload['file_path']
+        self.log.debug("Parsing file: %s", path)
 
-        response = await ParsingProcessorManager.operation_parser(file_path)
-        json_msg = json.dumps({"env_list": response})
+        model = await self.operation_parser(path)
 
         self.set_status(200)
-        self.set_header("Content-Type", 'application/json')
-        self.finish(json_msg)
-        self.flush()
-
-
-class ParsingProcessorManager(SingletonConfigurable):
-
-    async def operation_parser(self, file_path):
-        res = await asyncio.get_event_loop().run_in_executor(
-            None, self.parse_operation_envs, file_path)
-        return res
+        # TODO: Validation of model
+        self._finish_model(model)
 
-    @staticmethod
-    def parse_operation_envs(operation_filepath: str) -> List:
+    async def operation_parser(self, operation_filepath):
         """
-        Given the path to a File, will return a list of parsed environmental variables
-        found in File
+        Given the path to a File, will return a dictionary model
         :param operation_filepath: absolute path to a File to be parsed
-        :return: a list of environmental variables
+        :return: a model dict
         """
         operation = OperationParser.get_instance(filepath=operation_filepath)
-        env_list = operation.env_list()
+        model = operation.model
```

Comment:
I think we might want to still keep the method specific to the target - envs in this case.  Then, as other targets are added, they would be collected here.  Once the targets have been collected, THEN the response model is constructed.  We may even find that not all targets are always contained within the operation itself - some targets may require access to other objects relative to the operation.